recriando todo o docker para comportar o postgres

hesfra opened the pull request:
retirando tudo relacionado ao mongo, adicionando o postgres com um script de criação de tabela assim que o docker rodar!

hesfra committed:
criando o  docker

michaelrodriguess committed:
Merge pull request #1 from michaelrodriguess/feat/newDocker

Feat/new docker

hesfra committed:
recriando todo o docker para comportar o postgres

didibrabosa commented on docker-compose.yml:
```diff
@@ -1,35 +1,29 @@
 services:
   web:
-    build:
+    build: 
```
tem um espaço ali, tenho toque mano

hesfra replied:
tu já organizou

didibrabosa reviewed (approved):
the brabo!

AurelioNM commented on docker-compose.yml:
```diff
@@ -1,35 +1,29 @@
+      - db  # Certifique-se de que o banco de dados esteja disponível antes do aplicativo web iniciar
     networks:
       - customers_service_network
+  db:
```
`container_name: customer-database`

hesfra replied:
feito!

AurelioNM commented on docker-compose.yml:
```diff
@@ -1,35 +1,29 @@
-version: "3.8"
-
 services:
   web:
```
Daqui a pouco vão ter vários containers de diferentes services rodando na sua maquina. Sempre bom colocar um nome neles pra não se perder
`container_name: customer-service`

hesfra replied:
feito!

AurelioNM commented on docker-compose.yml:
```diff
@@ -1,35 +1,29 @@
-      MONGO_INITDB_ROOT_PASSWORD: password123 # Senha root
-    volumes:
-      - mongo_data:/data/db # Volume persistente para os dados do MongoDB
+      - "5432:5432"
```
Se todo mundo colocar na porta padrão, vai dar ruim né?
Parta do pressuposto que o mamífero já deve ter algum PG rodando na 5432
Troca a porta, e avisa no grupo, pros proximos só irem incrementando

hesfra replied:
feito!

AurelioNM commented on main.py:
```diff
@@ -11,33 +12,39 @@ class User(BaseModel):
+        await conn.close()
+
+
+# Função para criar um novo usuário
```
Arranca esses comentarios que não adicionam em nada

hesfra replied:
feito!

AurelioNM commented on scripts/init_db.sql:
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE IF NOT EXISTS users (
+    id SERIAL PRIMARY KEY,
```
ID na vida real é UUID ou ULID
No C6 padrão é ULID, VARCHAR(26)

A aplicação gerando o ULID, não o DB
Pesquisem pq não é uma boa prática deixar o DB criar o ULID

hesfra replied:
feito, também vai ser feita uma função de criação de ids!

AurelioNM commented on scripts/init_db.sql:
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE IF NOT EXISTS users (
```
O nome da aplicação é CUSTOMER-service
E a tabela é users??

Raph2ll replied:
Nem todo user é customer.

hesfra replied:
feita a alteração já!

AurelioNM commented on settings.py:
```diff
@@ -0,0 +1,20 @@
+
+class Settings(BaseSettings):
+    DATABASE_HOST: str = (
+        "db"  # O host do banco de dados (por exemplo, "db" para o container Docker)
```
Esses codigo de chatgpt, cheio de comentario...

hesfra replied:
retirado todos os comentários

AurelioNM commented on scripts/init_db.sql:
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE IF NOT EXISTS users (
+    id SERIAL PRIMARY KEY,
+    name TEXT NOT NULL,
+    email TEXT NOT NULL
```
Tem noção do tamanho q é um campo TEXT??
A menos q vc esteja querendo escrever uma bíblia aqui, eu trocaria pra VARCHAR com tamanho limitado

Senti falta de Index na tabela tbm...
E de colunas de timestamp de criaçao e update

hesfra replied:
alterado os tipos do banco!

AurelioNM reviewed (changes requested):
Melhore

Raph2ll committed:
fix: removendo comentários desnecessários

Raph2ll committed:
fix: Ajustando a tipagem dos dados na tabela users.

Raph2ll committed:
fix: Alterando nome do container db

Arthuralentejo commented on settings.py:
Aqui vc precisa usar dot env pra n ter suas credencias do banco "hard coded" 
https://pypi.org/project/python-dotenv/
https://medium.com/@habbema/dotenv-9915bd642533

hesfra replied:
adicionado o .env!

hesfra committed:
organizando o merge

hesfra committed:
Changes to be committed:
	modified:   Dockerfile
	modified:   docker-compose.yml
	modified:   main.py
	modified:   scripts/init_db.sql
	modified:   settings.py

hesfra commented:
foram feitas todas as alterações solicitadas, vou fechar esse pr e abrir um novo com as novas configs

hesfra committed:
adicionando o dotenv

michaelrodriguess commented on main.py:
```diff
@@ -11,33 +12,41 @@ class User(BaseModel):
-    raise HTTPException(status_code=404, detail="User not found")
+@app.get("/users")
+async def get_users():
+    return "hello world"
```
Just putting it on record here, this was done for testing I suppose, then we'll change the arc of the project.

hesfra replied:
yeah, that's right!!!

hesfra committed:
retirando comentário

michaelrodriguess commented on scripts/init_db.sql:
```diff
@@ -0,0 +1,16 @@
+    postal_code VARCHAR(20) NOT NULL,
+    country VARCHAR(100) NOT NULL,
+    FOREIGN KEY (customer_id) REFERENCES customers(id) ON DELETE CASCADE
+);
```
here too, I believe that this data structure is initial and that we will soon change. Maybe it makes sense to put versions for these .sql

hesfra replied:
this delete on cascade is used to delete the address automatically after the customer delete

didibrabosa reviewed (approved):
lgtm

hesfra committed:
reorganizando o projeto

hesfra committed:
Merge branch 'feat/mongoToPostgres' of github.com:michaelrodriguess/customer-service into feat/mongoToPostgres

hesfra committed:
removido o settings, adicionado o .env, estruturada nova pasta para o storage, pra onde o script foi movido

hesfra commented:
todo o código foi testado, com 2 requisições de get e post, utilizando cursor, existe um problema na criação do ON  UPDATE que o postgres não reconhece e não utiliza, portanto o update deve contemplar um timestamp com o current_time para que seja feita a alteração no banco, o arquivo main.py foi alterado, para apenas comportar a inicialização do fast api.

didibrabosa commented on storage/scripts/init_db.sql:
```diff
@@ -0,0 +1,9 @@
+    updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+);
+
+CREATE UNIQUE INDEX customer_email on customers(email);
```
wouldn't it be better to pass the "UNIQUE NOT NULL" value directly into the email field in the table?

didibrabosa committed:
git commit -m "refactor: update init_db.sql to set 'updated_at' as TIMESTAMP NULL and rename index to 'customer_email_idx'"

Raph2ll commented on scripts/init_db.sql:
```diff
@@ -0,0 +1,16 @@
+CREATE TABLE IF NOT EXISTS customers (
+    id VARCHAR(26) PRIMARY KEY,
```
@hesfra  Esse campo precisa ser CHAR(26), um ULID nunca vai ser maior ou menor que 26.

hesfra replied:
feita a  alteração!

hesfra committed:
alterando o ID no banco para char

hesfra committed:
Merge branch 'feat/mongoToPostgres' of github.com:michaelrodriguess/customer-service into feat/mongoToPostgres
merging to change the varchar to char in db

didibrabosa reviewed (commented):
nice one guys!
